src/VisualStudio/Core/Def/Implementation/TaskList/ExternalErrorDiagnosticUpdateSource.cs

```diff
@@ -300,13 +317,12 @@ private async Task SyncBuildErrorsAndReportOnBuildCompletedAsync(DiagnosticAnaly
             var solution = inProgressState.Solution;
             var (allLiveErrors, pendingLiveErrorsToSync) = await inProgressState.GetLiveErrorsAsync().ConfigureAwait(false);
 
-            // Make those errors live errors
-            await diagnosticService.SynchronizeWithBuildAsync(_workspace, pendingLiveErrorsToSync, onBuildCompleted: true).ConfigureAwait(false);
-
-            // Raise events for ones left-out, i.e. build only errors.
+            // Raise events for build only errors
             var buildErrors = GetBuildErrors().Except(allLiveErrors).GroupBy(k => k.DocumentId);
             foreach (var group in buildErrors)
             {
+                inProgressState.CancellationToken.ThrowIfCancellationRequested();
```

Comment:
why is hte cancellation token part of the inprogressstate rather than something ambiently passed through?  not complainingly, just trying to understand the abstraction.

Reply:
I was just being lazy, fixed it properly to have token passed along with all async API calls now.

Reply:
no worries :)  note: i tend to vasdtly prefer that cancellation tokens not be captured as state, unless that is truly the right design for the owrk.  It's super super super easy for incorrect cancellation excpetions to happen when we do this and for captures to end up doing very unexpected stuff.  So i'm glad that we can stick with the normal pattern here.